R-package/R/lgb.Booster.R

```diff
@@ -55,6 +55,7 @@ Booster <- R6::R6Class(
 
           # Create private booster information
           private$train_set <- train_set
+          private$train_set_version <- train_set$.__enclos_env__$private$version
```

Comment:
Maybe a naive question, but what does it mean to have a new "version" of the training set? Is it mainly updating weights as part of boosting? I think I am missing some context, so I don't understand what problem this patch solves.

Reply:
currently, Dataset is allowed to update label/weight/... (field data) during training, but the booster cannot capture these changes. And ideally, we should update related classes, such as the cached label/weight/.. in Objective/Metric. So I add a version number to indicate the changes of these fields.

Reply:
Ah I see ok, makes sense. Thanks!